input/input.go

```diff
@@ -44,31 +46,59 @@ func NewDefaultHandler(metrics mdata.Metrics, metricIndex idx.MetricIndex, input
 	}
 }
 
-// process makes sure the data is stored and the metadata is in the index
+// ProcessMetricPoint updates the index if possible, and stores the data if we have an index entry
 // concurrency-safe.
-func (in DefaultHandler) Process(metric *schema.MetricData, partition int32) {
-	if metric == nil {
+func (in DefaultHandler) ProcessMetricPoint(point schema.MetricPoint, partition int32) {
+	in.metricsReceived.Inc()
```

Comment:
I think this should be a separate metric.  it would be useful to see the ratio of metricPoint:MetricData being ingested

Reply:
right. will also do the same for the invalid metric